src/http_client_async.zig

```diff
@@ -1470,13 +1898,18 @@ pub fn onData(this: *HTTPClient, comptime is_ssl: bool, incoming_data: []const u
                 return;
             }
 
+            if (this.proxy_tunneling and this.proxy_tunnel == null) {
+                this.startProxyHandshake(is_ssl, socket);
+                return;
+            }
+
             if (body_buf.len == 0) {
                 return;
             }
 
             if (this.state.response_stage == .body) {
                 {
-                    const is_done = this.handleResponseBody(body_buf, true) catch |err| {
```

Comment:
I don't think this should be changed unless it's a proxy? This is a good optimization because it avoids an extra heap allocation for tiny HTTP responses which fit inside the stack

Reply:
It was a mistake, just reverting it right now